testinfra/backend/winrm.py

```diff
@@ -0,0 +1,66 @@
+# coding: utf-8
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import unicode_literals
+from __future__ import absolute_import
+
+from testinfra.backend import base
+
+try:
+    import winrm
+except ImportError:
+    raise RuntimeError((
+        "You must install the pywinrm package (pip install pywinrm) "
+        "to use the winrm backend"))
+
+import winrm.protocol
+
+
+class WinRMBackend(base.BaseBackend):
+    """Run command through winrm command"""
+    NAME = "winrm"
+
+    def __init__(self, hostspec, password=None, ssl=True, verify_ssl=True,
+                 transport=None, *args, **kwargs):
+        self.host, self.username, self.port = self.parse_hostspec(hostspec)
+        self.password = password
+        self.ssl = ssl
+        self.verify_ssl = verify_ssl
+        self.transport = transport or 'ntlm'
+        super(WinRMBackend, self).__init__(self.host, *args, **kwargs)
+
+    def run(self, command, *args, **kwargs):
+        return self.run_winrm(self.get_command(command, *args))
+
+    def run_winrm(self, command, *args):
+        conn_args = {
+            'endpoint': '{proto}://{host}:{port}/wsman'.format(
+                proto='https' if self.ssl else 'http',
+                host=self.host,
+                port=self.port
```

Comment:
What if we don't supply a port in ``hostspec`` ? It will be set to ``None`` and probably create a wrong ``endpoint`` here.

I'd suggest

```python
'endpoint': '{proto}://{host}{port}/wsman'.format(
    [...]
    port=':{}'.format(self.port) if port else ''
)
```

Reply:
looks good, will do